src/app/components/breadcrumbs/service/terra-breadcrumbs.service.ts

```diff
@@ -0,0 +1,426 @@
+        {
+            // you can set a name to update the breadcrumb name
+            if(!isNullOrUndefined(name))
+            {
+                breadcrumb.name = this.translation.translate(name);
+            }
+            // or it will be updated automatically from it's route data
+            else
+            {
+                breadcrumb.name = this.getBreadcrumbLabel(route);
+            }
+        }
+    }
+
+    private findBreadcrumbContainerByUrl(url:string):TerraBreadcrumbContainer
+    {
+        let urlPartsCount:number = url.split('/').length - 1;
+
+        let container:TerraBreadcrumbContainer = this._containers[urlPartsCount];
+
+        if(isNullOrUndefined(container))
+        {
+            console.error('No existing breadcrumb container found.');
+        }
+
+        return container;
+    }
+
+    private findBreadcrumbByUrl(url:string):TerraBreadcrumb
+    {
+        let container:TerraBreadcrumbContainer = this.findBreadcrumbContainerByUrl(url);
+
+        let routerLink:string = '/' + this._initialPath + url;
+
+        let breadcrumb:TerraBreadcrumb;
+
+        if(!isNullOrUndefined(container) &&
+           !isNullOrUndefined(container.currentSelectedBreadcrumb) &&
+           container.currentSelectedBreadcrumb.routerLink === routerLink)
+        {
+            breadcrumb = container.currentSelectedBreadcrumb;
+        }
+        else
+        {
+            breadcrumb = container.breadcrumbList.find((bc:TerraBreadcrumb) =>
+            {
+                return bc.routerLink === routerLink;
+            });
+        }
+
+        return breadcrumb;
+    }
+
+    public closeBreadcrumb(breadcrumbContainer:TerraBreadcrumbContainer, breadcrumb:TerraBreadcrumb):void
+    {
+        let breadcrumbList:Array<TerraBreadcrumb> = breadcrumbContainer.breadcrumbList;
+        let breadcrumbIndex:number = breadcrumbList.indexOf(breadcrumb);
+
+        // current selected breadcrumb should be closed?
+        if(breadcrumbContainer.currentSelectedBreadcrumb === breadcrumb)
+        {
+            let currentUrl:string = this.router.url;
+
+            let currentSelectedContainer:TerraBreadcrumbContainer = this._containers.find((bcc:TerraBreadcrumbContainer) =>
+            {
+                return bcc.currentSelectedBreadcrumb.routerLink === currentUrl;
+            });
+
+            // get the first remaining breadcrumb
+            breadcrumbContainer.currentSelectedBreadcrumb = breadcrumbIndex === 0 ? breadcrumbList[1] : breadcrumbList[0];
+
+            let currentSelectedIndex:number = this._containers.indexOf(currentSelectedContainer);
+            let breadcrumbContainerIndex:number = this._containers.indexOf(breadcrumbContainer);
+
+            // container has no more breadcrumbs
+            if(isNullOrUndefined(breadcrumbContainer.currentSelectedBreadcrumb))
+            {
+                this.removeBreadcrumbContainer(breadcrumbContainer);
```

Comment:
the breadcrumb containers are already removed in the `removeBreadcrumbsByParent`-Method

Reply:
@Neill83 Wie sieht das aus? Eigentlich kann `removeBreadcrumbContainer` doch weg oder nicht?